### PR TITLE
refactor!: remove popup wrapper class, use content classes for popover…

### PR DESCRIPTION
**Pull request by helenjer**
…s and tooltips

BREAKING CHANGE
Connected with https://github.com/gravity-ui/uikit/pull/656 and further discussion.

### CODE REVIEW

File: src/components/Popup/Popup.scss [hunk 1]
```diff
@@ -86,39 +19,39 @@ $transition-distance: 10px;
     }
 
     &_exit_active {
-        &[data-popper-placement*='bottom'] #{$block} {
+        &[data-popper-placement*='bottom'] #{#{$block}}__content {
```
Comment (amje) [inline src/components/Popup/Popup.scss:22]: double templating

File: src/components/Popup/Popup.scss [hunk 2]
```diff
@@ -151,6 +84,72 @@ $transition-distance: 10px;
             transform: rotate(-90deg);
         }
     }
+
+    &__content {
+        // Can be overridden, see Tooltip
+        --yc-popup-background-color: var(--yc-color-base-float);
+        --yc-popup-border-color: var(--yc-color-line-solid);
+        --yc-popup-border-width: 1px;
```
Comment (amje) [inline src/components/Popup/Popup.scss:92]: These variables should remain on the root node

File: src/components/Tooltip/Tooltip.scss [hunk 3]
```diff
@@ -3,17 +3,19 @@
...
-    // prevent glitch between two nearby tooltip refs
-    pointer-events: none;
+    &__popup-content {
+        // prevent glitch between two nearby tooltip refs
+        pointer-events: none;
+
+        // [class] for increasing specificity
+        &[class] {
+            --yc-popup-border-width: 0;
+            --yc-popup-background-color: var(--yc-color-base-float-heavy);
```
Comment (amje) [inline src/components/Tooltip/Tooltip.scss:13]: Same as above

**Review by amje — CHANGES_REQUESTED** [review]

File: src/components/Popup/Popup.scss (hunk 1 above)
Comment (helenjer) [reply src/components/Popup/Popup.scss]: fixed

File: src/components/Tooltip/Tooltip.tsx [hunk 4]
```diff
@@ -15,6 +15,7 @@ export interface TooltipProps extends TooltipDelayProps {
     children: React.ReactElement;
     className?: string;
     contentClassName?: string;
+    popupContentClassName?: string;
```
[the remark this answers is not in the document]
Comment (helenjer) [reply src/components/Tooltip/Tooltip.tsx]: It was added for backward ability to customize popup content as popup className previously was used for it (content) now for top-level elem (wrapper). Guess we can omit due to major, I've removed.

File: src/components/Popup/Popup.scss (hunk 2 above)
Comment (helenjer) [reply src/components/Popup/Popup.scss]: I've moved it up to #{$block}

File: src/components/Popover/Popover.scss [hunk 5]
```diff
@@ -122,7 +124,7 @@ $block: '.#{variables.$ns}popover';
 
             &_special {
                 // Increasing specificity for overrides
-                &#{&} {
+                #{$class}__tooltip-popup-content {
```
[the remark this answers is not in the document]
Comment (helenjer) [reply src/components/Popover/Popover.scss]: returned back

File: src/components/Tooltip/Tooltip.scss (hunk 3 above)
Comment (helenjer) [reply src/components/Tooltip/Tooltip.scss]: Moved up to the #{$block}

**Review by amje — APPROVED** [review]

**Pull request merged**